stdlib/src/collections/dict.mojo

```diff
@@ -444,9 +445,14 @@ struct Dict[K: KeyElement, V: CollectionElement](
         """Initialize an empty dictiontary."""
         self.size = 0
         self._n_entries = 0
-        self._reserved = Self._initial_reservation
-        self._index = _DictIndex(self._reserved)
-        self._entries = Self._new_entries(self._reserved)
+        self._entries = Self._new_entries(Self._initial_reservation)
+        self._index = _DictIndex(len(self._entries))
```

Comment:
Nit: I guess it makes more sense to use `Self._initial_reservation` here to make clear that it's compile time constant?

Reply:
`len(self._entries)` can be greater than or equal to `Self._initial_reservation`, but they're not equal in all cases. See the section "Using all the capacity available in the List" of the pull request description

Reply:
My point exactly: should we leave that to `List` at all? Your call anyways, shouldn't be blocking.